ldcbench.api/src/main/java/org/dice_research/ldcbench/graph/GrphBasedGraph.java

```diff
@@ -253,4 +263,17 @@ public void setGraphIdOfNode(int nodeId, int graphId, int externalId) {
     public String toString() {
         return graph.toString();
     }
+
+    @Override
+    public void addBlankNodes(int nodeCount, long seed) {
+        blankNodesIndex = this.getNumberOfNodes();
+        Random generator = new Random(seed);
+        addNodes(nodeCount);
+        for (int i = blankNodesIndex; i < this.getNumberOfNodes(); i++) {
+            //Get a random Source Node
+            int sourceNode = generator.nextInt(blankNodesIndex);
+            //Set the proper type?
+            addEdge(sourceNode, i, 0);
+        }
+    }
```

Comment:
It is not the task of the graph to randomly add a blank node. This class is only responsible to store data about the graph and (as GraphBuilder instance) interact with the graph by adding and removing certain nodes or edges. 

However, the functionality you implemented above goes beyond that as you implement a certain algorithm to choose other nodes to which you can add the blank nodes. You also hard coded the property ID of the edge. All of this are things that are part of the overall generation algorithm implementation and shouldn't be in this Graph class.

Please note that the implementation seems to be fine. It is only in the wrong class ;)
Feel free to create a "BlankNodeAdder" with this piece of code. This class could then be used by all graph generator classes.